src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/UpdateContactDetailsContestedSubmittedHandler.java

```diff
@@ -0,0 +1,137 @@
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.EventType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CaseDocument;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CaseType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseData;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseDetails;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.OnlineFormDocumentService;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.noc.nocworkflows.UpdateRepresentationWorkflowService;
+
+import java.util.Optional;
+
+import static uk.gov.hmcts.reform.finrem.caseorchestration.OrchestrationConstants.YES_VALUE;
+
+@Slf4j
+@Service
+public class UpdateContactDetailsContestedSubmittedHandler extends FinremCallbackHandler {
+
+    private final UpdateRepresentationWorkflowService nocWorkflowService;
+
+    private final OnlineFormDocumentService service;
+
+    public UpdateContactDetailsContestedSubmittedHandler(FinremCaseDetailsMapper finremCaseDetailsMapper,
+                                                         UpdateRepresentationWorkflowService nocWorkflowService,
+                                                         OnlineFormDocumentService service) {
+        super(finremCaseDetailsMapper);
+        this.nocWorkflowService = nocWorkflowService;
+        this.service = service;
+    }
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, CaseType caseType, EventType eventType) {
+        return CallbackType.ABOUT_TO_SUBMIT.equals(callbackType)
+            && CaseType.CONTESTED.equals(caseType)
+            && EventType.UPDATE_CONTACT_DETAILS.equals(eventType);
+    }
+
+    @Override
+    public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(FinremCallbackRequest callbackRequest,
+                                                                              String userAuthorisation) {
+        FinremCaseDetails finremCaseDetails = callbackRequest.getCaseDetails();
+        log.info("Invoking contested {} submitted callback for Case ID: {}", callbackRequest.getEventType(), finremCaseDetails.getId());
+
+        FinremCaseData caseData = finremCaseDetails.getData();
+
+        removeApplicantSolicitorDetails(caseData);
+        removeRespondentDetails(caseData);
+
+        if (caseData.isAppAddressConfidential() || caseData.isRespAddressConfidential()) {
+            CaseDetails caseDetails = finremCaseDetailsMapper.mapToCaseDetails(callbackRequest.getCaseDetails());
+            CaseDocument document = service.generateContestedMiniFormA(userAuthorisation, caseDetails);
+            caseData.setMiniFormA(document);
+        }
+
+        if (Optional.ofNullable(caseData.getContactDetailsWrapper().getUpdateIncludesRepresentativeChange()).isPresent()
+            && caseData.getContactDetailsWrapper().toString().equals(YES_VALUE)) {
```

Comment:
There is a bug here.
 `&& caseData.getContactDetailsWrapper().toString().equals(YES_VALUE)) {`
Presumably it should be checking getUpdateIncludesRepresentativeChange.